blinky-statik/src/main/java/org/spideruci/analysis/statik/flow/StatikFlowGraph.java

```diff
@@ -84,31 +100,36 @@ public void buildIcfg() {
     ArrayList<SootMethod> visited = new ArrayList<>();
     
     while(!worklist.isEmpty()) {
-      SootMethod src = worklist.remove(0);
-      if(methodIsInvalid(src)) {
+      SootMethod srcMethod = worklist.remove(0);
+      if(methodIsInvalid(srcMethod)) {
         continue;
       }
 
-      visited.add(src);
-      icfgMgr.addSootMethodToIcfg(src);
-      Items<Edge> outEdges = new Items<>(callgraph.edgesOutOf(src));
-      
-      
+      visited.add(srcMethod);
+      icfgMgr.addSootMethodToIcfg(srcMethod);
+      Items<Edge> outEdges = new Items<>(callgraph.edgesOutOf(srcMethod));
+
       for(Edge edge : outEdges) {
-        SootMethod tgt = edge.tgt();
+        SootMethod tgtMethod = edge.tgt();
         
-        if(methodIsInvalid(tgt))
+        if(methodIsInvalid(tgtMethod))
           continue;
         
-        Body body = tgt.retrieveActiveBody();
+        Body tgtBody = tgtMethod.retrieveActiveBody();
         
         Unit callUnit = edge.srcUnit();
-        Unit entryUnit = body.getUnits().getFirst();
+        Unit entryUnit = tgtBody.getUnits().getFirst();
         
-        icfgMgr.addIcfgEdge(callUnit, src, entryUnit, tgt);
+        List<Unit> tgtExitUnits = Statik.GET_UNIT_GRAPH(tgtMethod).getTails();
```

Comment:
this change and the general additions related to `List<Unit> tgtExitUnits` actually hooks up method exits (instructions/units with no successors), to the original call site. I was not doing this before.